webapp/channels/src/utils/markdown/apply_markdown.ts

```diff
@@ -492,6 +491,68 @@ export function applyLinkMarkdown({selectionEnd, selectionStart, message, url =
     };
 }
 
+function applyCodeMarkdown({selectionEnd, selectionStart, message}: ApplySpecificMarkdownOptions) {
+    if (isSelectionMultiline(message, selectionStart, selectionEnd)) {
+        return applyCodeBlockToSelection({selectionEnd, selectionStart, message});
+    }
+    return applyMarkdownToSelection({selectionEnd, selectionStart, message, delimiter: '`'});
+}
+
+const applyCodeBlockToSelection = ({
```

Comment:
For this function, is the only difference between this and `applyMarkdownToSelection` that this has separate start and end delimiters?

If so, it'd be nice to not duplicate most of this logic, so could we modify `applyMarkdownToSelection` to support either `delimiter` or `delimiterStart`/`delimiterEnd` as arguments? I think that wouldn't be too hard if you just use this version and set them like `delimiterStart = options.delimiterStart ?? options.delimiter`

Reply:
Yes, it's the only difference, I added a new function because of the comment before the switch case:
```
    /**
     * all options that need to be handled in a ver specific way have their own applyMarkdown (sub-)functions.
     * The rest just define their delimiters and return the basic applyMarkdownToSelection function.
     */
```
I'll change the code to only use applyMarkdownToSelection 👍